src/modules/seasons/seasons.service.ts

```diff
@@ -1,16 +1,41 @@
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
-import { Season } from '../../entities/season.entity';
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+import { InternalServerErrorException } from '@nestjs/common/exceptions';
+import { SeasonEntity } from 'src/entities/season.entity';
+import { CreateSeasonDto, SeasonDto } from './seasons.types';
 
 @Injectable()
 export class SeasonsService {
     constructor(
-        @InjectRepository(Season)
-        private seasonsRepository: Repository<Season>
-    ) {}
+        @InjectRepository(SeasonEntity)
+        private seasonsRepository: Repository<SeasonEntity>,
+        @InjectMapper() private readonly classMapper: Mapper,
+    ) { }
 
-    findAll(): Promise<Season[]> {
-        return this.seasonsRepository.find();
+    async findAll(): Promise<SeasonDto[]> {
+        try {
+            return this.classMapper.mapArrayAsync(await this.seasonsRepository.find(), SeasonEntity, SeasonDto);
+        } catch {
+            throw new InternalServerErrorException('could not retrieve seasons');
```

Comment:
small note, it's idiomatic to not provide a custom error message with `internal server error` type errors - we'd generally just log them to our logging service (which we have yet to implement).